Makefile

```diff
@@ -0,0 +1,10 @@
+# Experimental, haven't verified yet
+CXX:=clang++
+CXXFLAGS:=-Wall -Wextra -Isrc/ -std=c++17 -ggdb
+CXXLIBS:=-lSDL2 -lSDL2_ttf
+SRCFILES := $(shell find src/ -type f -name "*.cpp")
+
+.PHONY: build
+build:
+	$(CXX) $(CXXFLAGS) $(CXXLIBS) $(SRCFILES) -o build/editor
```

Comment:
I think the whole point of make is to be able to parallelize compilation and do it incrementally so compiling everything at once probably isn't very useful.